metaseq/benchmark/generator.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+import random
+import socket
+import sys
+from typing import Tuple
+
+import numpy as np
+import torch
+from torch.profiler.profiler import (
+    ProfilerActivity,
+    schedule,
+    tensorboard_trace_handler,
+)
+from tqdm import tqdm
+
+from metaseq import options
+from metaseq.dataclass.configs import MetaseqConfig
+from metaseq.dataclass.utils import convert_namespace_to_omegaconf
+from metaseq.distributed import utils as distributed_utils
+from metaseq.hub_utils import GeneratorInterface
+
+logging.basicConfig(format="%(asctime)s | %(name)s | %(message)s", level=logging.INFO)
+logger: logging.Logger = logging.getLogger("metaseq.benchmark.generator")
+
+WARMUP_STEPS = 5
+NUM_REPEATS = 5
+generator = None
+
+
+def main(cfg: MetaseqConfig) -> None:
+    global generator
+    torch.manual_seed(random.randint(1, 20000))
+
+    logger.info("Instantiating a generator interface and loading a model")
+    generator = GeneratorInterface(cfg)
+    generator.load_model()
+
+    batch_size = cfg.dataset.batch_size
+    input_length = cfg.task.max_source_positions
+    output_length = cfg.task.max_target_positions
```

Comment:
Lol funny names for input and output length. From fairseq I'm guessing

Reply:
I think these options are closest to input and output lengths. For context, it's nontrivial to introduce new arguments as we have some validation logic related dataclass fields, which I don't think should be modified for this purpose.